### PR TITLE
IDEMPIERE-4885 Possible bug with uom conversion on invoice when create from shipment

### PR DESCRIPTION
**Pull request by alara123**
We correct the issue keeping the unit from the in/out record and we are only converting the price of the item.

[https://idempiere.atlassian.net/browse/IDEMPIERE-4885](https://idempiere.atlassian.net/browse/IDEMPIERE-4885)

### CODE REVIEW

File: org.adempiere.ui/src/org/compiere/grid/CreateFromInvoice.java [hunk 1]
```diff
@@ -547,6 +547,14 @@ else if (M_RMALine_ID != 0)
 				if (inoutLine != null)
 				{
 					invoiceLine.setShipLine(inoutLine);		//	overwrites
+					if(invoiceLine.getC_UOM_ID()!=inoutLine.getC_UOM_ID()) {
+						invoiceLine.setC_UOM_ID(inoutLine.getC_UOM_ID());						
+						BigDecimal PriceEntered = MUOMConversion.convertProductFrom (invoice.getCtx(), M_Product_ID, 
+								inoutLine.getC_UOM_ID(), invoiceLine.getPriceEntered());
+							if (PriceEntered == null)
+								PriceEntered = invoiceLine.getPriceEntered();
```
Comment (hengsin) [inline org.adempiere.ui/src/org/compiere/grid/CreateFromInvoice.java:555]: this isn't a safe fallback, better to raise error here if conversion is not possible.

**Review by hengsin — CHANGES_REQUESTED** [review]
please fix the price conversion fallback

File: org.adempiere.ui/src/org/compiere/grid/CreateFromInvoice.java (hunk 1 above)
Comment (alara123) [reply org.adempiere.ui/src/org/compiere/grid/CreateFromInvoice.java]: code updated for throw error when no conversion found

**Comment by alara123** [comment]
code updated for throw error when no conversion found

**Review by hengsin — APPROVED** [review]
looks good.

**Pull request merged**